### ISSUE TITLE
Compilation fails inside VS with "Unnamed types are not supported"  in presence of a JS method with `Memory<byte>` argument

### ISSUE THREAD

**Issue by ceztko** [microsoft/node-api-dotnet#387]
In a NAOT enabled project, the regular .NET compilation (not the AOT publish) fails **inside Visual Studio (only)** if a  `Memory<byte>` argument is used in a JS exported method. Accordingly to [this](https://github.com/microsoft/node-api-dotnet/issues/208#issuecomment-1934670450) message,  `Memory<byte>` can be used to project a `UInt8Array` parameter in JS, and this works perfectly with a compilation issued from the command line with `dotnet build` or `dotnet publish` . Instead, the regular .NET compilation inside Visual Studio fails with the following generator related error:

```
Error (active)	NAPI1001	NotSupportedException : Unnamed types are not supported: void*
at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.AsType(ITypeSymbol typeSymbol, Type[] genericTypeParameters, Boolean buildType)
at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildSymbolicConstructor(TypeBuilder typeBuilder, IMethodSymbol constructorSymbol, Type[] genericTypeParameters)
at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildSymbolicTypeMembers(ITypeSymbol typeSymbol, TypeBuilder typeBuilder, Type[] genericTypeParameters)
at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildSymbolicObjectType(INamedTypeSymbol typeSymbol, String typeFullName, Type[] genericTypeParameters, Boolean buildType)   at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.AsType(ITypeSymbol typeSymbol, Type[] genericTypeParameters, Boolean buildType)
at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildSymbolicMethod(TypeBuilder typeBuilder, IMethodSymbol methodSymbol, Type[] genericTypeParameters)   at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildSymbolicTypeMembers(ITypeSymbol typeSymbol, TypeBuilder typeBuilder, Type[] genericTypeParameters)	aot-npm-package	E:\node-api-dotnet\examples\aot-npm-package\lib\CSC	1		
```

Steps to reproduce:
- Clone https://github.com/microsoft/node-api-dotnet/tree/main and move to `examples/aot-npm-package`
- Run `dotnet pack ../..`
- Open `lib/aot-npm-package.csproj` with Visual Studio and replace `Example.cs` with the following content:

```
using System;

namespace Microsoft.JavaScript.NodeApi.Examples;

[JSExport]
public static class Example
{
    public static void Foo(Memory<byte> bytes)
    {

    }
}
```
- Build the solution (F6)
- -> Error.

**Comment by jasongin** [comment]
Fixed in version `0.8.15`.

**Comment by ceztko** [comment]
Thank you so much! We are so happy we chose node-api-dotnet.
